src/main/java/com/redhat/devtools/lsp4ij/settings/ui/ValidatableConsoleWidget.java

```diff
@@ -0,0 +1,43 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Red Hat Inc. and others.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Mitja Leino <mitja.leino@hotmail.com> - Initial API and implementation
+ *******************************************************************************/
+package com.redhat.devtools.lsp4ij.settings.ui;
+
+import com.intellij.openapi.ui.ValidationInfo;
+import com.intellij.ui.JBColor;
+import com.intellij.util.ui.JBUI;
+
+import javax.swing.*;
+import java.util.List;
+import java.awt.*;
+
+/**
+ * A shared interface meant to simplify creating validatable components used in
+ * NewLanguageServerDialog and LanguageServerView (LSP console)
+ */
+public interface ValidatableConsoleWidget {
+    /**
+     * Set a common error border for the widget
+     * @param jComponent interface implementor (e.g. setErrorBorder(this);)
+     */
+    default void setErrorBorder(JComponent jComponent) {
```

Comment:
setErrorBorder is more an utility class than an API.

Here some suggestion:

 * remove `setErrorBorder `method
 * add a new void `updateUi(boolean hasError)` method which is called after the call of validate (hasError must be comuted by coparing the size of validations before/after the cal of validate.
 * add a new method `initializeValidator(ValidateDialog)` which add the proper listener (which is done today in the addValidator)